hack/generate/samples/pkg/utils.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkg
+
+import (
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// CheckError will check the error and exit with 1 when as errors
+func CheckError(err error) {
+	if err != nil {
+		log.Errorf("error to generate the sample: %s", err)
+		os.Exit(1)
+	}
```

Comment:
This makes it very difficult to figure out where the error is. While you are "not repeating yourself" but what this means is whenever the thing fails I will see "error to generate the sample: SOMEERROR MESSAGE" I will search the code and find line 26 of `utils.go` but it won't tell me WHERE the error actually happened. 

I would prefer one of the following changes:

1. just do the `if err != nil` check directly in the *sample.go file with a meaningful error message. 
```go
	log.Infof("creating the project")
	err = mh.ctx.Init(
		"--plugins", "helm",
		"--domain", mh.ctx.Domain)
	if err != nil {
		log.Errorf("error creating the project: %s", err)
		os.Exit(1)
	}
```

2. if you really want to DRY then change `CheckError` to take in a message.
```go
func CheckError(msg string, err error) {
	if err != nil {
		log.Errorf("error %s: %s", msg, err)
		os.Exit(1)
	}
}
```
Then in the sample you can call it like this. 

```go
	log.Infof("creating the project")
	err = mh.ctx.Init(
		"--plugins", "helm",
		"--domain", mh.ctx.Domain)
	pkg.CheckError("creating the project", err)
```

This ensures that if something breaks, I can grep the message string and find exactly where the error came from.